# Solve all problems while demonstrating each step clearly. The Quiz is worth a  total of 10 points

## 1. Suppose an asset has the following factor sensitivities: $\beta_{1} = 0.5, \beta_{2} = −0.3, \beta_{3} = 0.7$. If the expected returns of the three factors are $8 \%,\ 6\%,$ and $10 \%$ respectively, and the risk-free rate is $4\%$ calculate the expected return of the asset according to arbitrage pricing theory.

$$
E(R_i)-\ expected\ return\ of\ an\ asset
$$
$$
R_f-\ risk\ free\ rate
$$
$$
\beta_{1},\ \beta_{2},\ \beta_{3} -\ sensitivity\ of\ asset
$$
$$
E(F_1), E(F_2), E(F_3)-\ expected\ returns\ of\ the\ assets
$$

$$
E(R_i) = R_f + B_1 * E(F_1) + B_2 * E(F_2) + B_3 * E(F_3)
$$
$$
= \frac{4}{100} + 0.5 * \frac{8}{100} + (-0.3 *\frac{6}{100}) + 0.7 * \frac{10}{100}
$$

In [25]:
(0.5 *(8/100))

0.04

In [26]:
round((-0.3 * (6/100)), 4)

-0.018

In [28]:
round(0.7 * (10/100), 4)

0.07

In [30]:
0.04 + 0.04 - 0.018 + 0.07

0.132

$$
ans\ =\ 0.132
$$

## 2. Consider a portfolio consisting of two assets: Asset A and Asset B. Asset A has an expected return of $10\%$ and a standard deviation of $15\%$, while Asset B has an expected return of $8\%$ and a standard deviation of $10\%$. If you allocate $60\%$ of your portfolio to Asset A and $40\%$ to Asset B, what is the expected return and variance of your portfolio?

$$
E(R_A),\ E(R_B)- expected\ returns\ asset\ A\ and\ B\ respectively
$$
$$
w_a,\ w_b - weight\ of\ asset\ a\ and\ b
$$
$$
E(R_p)-\ expected\ return\ of\ portfolio
$$
$$
E(R_p) = w_a * E(R_A) + w_b * E(R_B)
$$

$$
= 0.6 * 0.1 + 0.4 * 0.08
$$

In [6]:
0.6 * 0.1 + 0.4 * 0.08

0.092

$
E(R_p) = 0.092 = 9.2 \%
$

$$
var(R_p) = w_a * var(A) + w_b * var(B) + w_a * w_b * cov(A, B) 
$$

$$
cov(A, B) = 0
$$
$$
var(A) = sd(A) ^ 2, var(B) = sd(B) ^ 2
$$
$$
var(A) = (0.6 * 0.15) ^2
$$
$$
var(B) = (0.4 * 0.1) ^2
$$

In [31]:
((0.6 * 0.15) **2) + ((0.4*0.1) **2) + 0

0.0097

$var(R_p) = 0.0097$

## 3. Given a corporate bond with a yield of 6% and a similar maturity risk-free rate of 3%, calculate the credit spread.


$$
credit\ spread = yield - risk-free\ rate
$$
$$
= 0.06 - 0.03
$$
$credit\ spread = 3%$

## 4. Calculate the credit valuation adjustment (CVA) for a portfolio of derivative contracts with a total notional value of $10 million, a default probability of 1%, and an LGD of 40%.

$$
cVA = Exposure * Default\ probability * LGD
$$
$$
= 10, 000, 000 * 0.01 * 0.4
$$

In [8]:
10_000_000 * 0.01 * 0.4

40000.0

$credit\ valuation\ adjustment = 40,000$

## 5. Calculate the Sharpe ratio for a portfolio with an average annual return of 10% and a standard deviation of returns of 15%. The risk-free rate is 3%.

$$
E(R_p)-\ is\ the\ expected\ return\ of\ the\ portfolio.
$$
$$
𝑅_𝑓-\ is\ the\ risk-free\ rate.
$$
$$
\sigma -is\ the\ standard\ deviation\ of\ the\ portfolio’s\ returns.
$$
$$
sharpe\ ratio = \frac{E(R_p) - R_f}{\sigma}
$$

$$
= \frac{0.1 - 0.03}{0.15}
$$

In [9]:
(0.1 - 0.03)/0.15

0.46666666666666673

$sharpe\ ratio = 0.466667$

## 6. Calculate the beta of a stock with an average return of 12% and a standard deviation of returns of 18%. The average return of the market is 8%, and its standard deviation of returns is 12%.


the stock has an internal correlation of 1 with the market. 

ie. The growth of the stock is directly dependent on the market

$Cor(R_i, R_m) = 1$

$\therefore$ 

$$
\beta (sensitivity) = \frac{Cov(R_i\ (stock's\ returns) - R_m\ (market returns))}{var(R_M)}
$$

$$
\beta = Cor(R_i, R_m) * \frac{\sigma_i}{\sigma_m}
$$

$$
sensitivity = \frac{0.18}{0.12}
$$

In [10]:
0.18/0.12

1.5

$sensitivity = 1.5$

## 7. Consider a hypothetical portfolio with the following thirty historical returns data:
|Day| 1| 2| 3| 4| 5| 6| 7| 8| 9| 10| 11| 12| 13| 14| 15|
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:
|Return (%)| -1.2| 0.5| 1.0| -0.7| -1.5| 2.3| -0.4| 0.6| -2.1| 1.2| 0.8| -1.1| 1.5| -0.3| 2.0|

$$
Table\ 1:\ Historical\ Returns\ Data\ (Days\ 1-15)|$$

|Day| 16| 17| 18| 19| 20| 21| 22| 23| 24| 25| 26| 27| 28| 29| 30|
|:---|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|---:|
|Return (%)| -1.7| 0.4| -0.5| 1.8| -2.2| 0.9| 1.1| -1.6| 2.4| -0.9| 1.3| -0.8| 0.7| -1.9| 1.4|

$$
Table\ 2:\ Historical\ Returns\ Data\ (Days\ 16-30)
$$

## Calculate the volatility of the returns and determine both the VaR and CVaR of the portfolio using the historical method on returns data, the parametric method, and Monte Carlo simulation. Comment on the results by the three methods and give a convincing justification



In [11]:
import numpy as np
returns = np.array([-1.2, 0.5, 1.0, -0.7, -1.5, 2.3, -0.4, 0.6, -2.1, 1.2, 0.8, -1.1, 1.5, -0.3, 2.0, -1.7,
                    0.4, -0.5, 1.8, -2.2, 0.9, 1.1, -1.6, 2.4, -0.9, 1.3, -0.8, 0.7, -1.9, 1.4])
returns = np.sort(returns)
returns

array([-2.2, -2.1, -1.9, -1.7, -1.6, -1.5, -1.2, -1.1, -0.9, -0.8, -0.7,
       -0.5, -0.4, -0.3,  0.4,  0.5,  0.6,  0.7,  0.8,  0.9,  1. ,  1.1,
        1.2,  1.3,  1.4,  1.5,  1.8,  2. ,  2.3,  2.4])

In [12]:
ret_elements = list(np.array(returns.shape))[0]
ret_elements

30

In [13]:
volatility = np.std(returns)
volatility

1.3650396819628847

### cVAR and VAR of the portfolio using Historical method

In [14]:
# using the confidence level of 95%
import math
loss_level_hist = math.ceil(0.05 * ret_elements)
loss_level_hist

2

In [24]:
var_historical = returns[:loss_level_hist, ][-1]
var_historical

-2.1

$VAR = -2.1$

In [16]:
cvar_historical = np.mean(returns[:loss_level_hist,])
cvar_historical

-2.1500000000000004

cVAR = -2.15

### cVAR and VAR of the portfolio using Parametric Method

In [17]:
mean_returns = np.mean(returns)
sd_returns = volatility
z_val = -1.65

$$
VAR = \mu + (z_{score} * \sigma)
$$

In [18]:
var_parametric = mean_returns + (z_val * sd_returns)
var_parametric

-2.1523154752387597

$$
cVAR = \mu + \frac{\phi (z_{score})*\sigma}{1 - C.L}
$$

$\phi$ - standard normalization of the pdf

In [19]:
from scipy.stats import norm
phi_data = norm.pdf(z_val)

cVAR_parametric = mean_returns - ( (phi_data * sd_returns)/ (1 - 0.95))
cVAR_parametric

-2.691913602055417

cVAR using parametric method = $-2.6191$

VAR using parametric method = $-2.1523$

## cVAR and VAR of the portfolio using Monte Carlo method

In [20]:
generated_random_data = np.sort(np.random.normal(mean_returns, sd_returns, 10_000))
generated_random_data

array([-5.42086659, -4.78354865, -4.77336518, ...,  4.94524895,
        4.9695628 ,  5.68176636])

In [21]:
element_at_5_percent = math.ceil(10000 * 0.05)
element_at_5_percent

500

In [32]:
var_monte_carlo = generated_random_data[:500,][-1]
var_monte_carlo

-2.2016173207825913

In [33]:
cvar_monte_carlo =  np.mean(generated_random_data[:element_at_5_percent,])
cvar_monte_carlo

-2.7230982715440586